service/raft/group0_fwd.hh

```diff
@@ -8,10 +8,52 @@
 
 #pragma once
 
-#include <iostream>
+#include <iosfwd>
+#include "raft/raft.hh"
+#include "gms/inet_address.hh"
 
 namespace service {
 
+// Address of a discovery peer
+struct discovery_peer {
+    raft::server_id id;
+    gms::inet_address ip;
+    bool operator==(const discovery_peer& rhs) const {
+        return id == rhs.id;
```

Comment:
Here you define equality using `id`, but then when you actually need to compare `discovery_peer`s, you create a separate `discovery_peer_equal` struct which compares using `ip`.

That made sense when we used `raft::server_address`es, which were normally compared using `raft::server_address`, but during discovery they were compared using `server_info`s.

But now, when we have a separate discovery peer struct, don't we always want to compare them using the IP? If so, `operator==` should use `ip` and we don't have to create a separate `discovery_peer_equal` struct inside `discovery`.

Reply:
Done.